extensions/shared/plan-utils.js

```diff
@@ -0,0 +1,111 @@
+/**
+ * External dependencies
+ */
+import { compact, get, startsWith } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { addQueryArgs } from '@wordpress/url';
+
+/**
+ * Internal dependencies
+ */
+import { isSimpleSite } from './site-type-utils';
+import getJetpackExtensionAvailability from './get-jetpack-extension-availability';
+import getSiteFragment from './get-site-fragment';
+
+/*
+ * Blocks list that require a paid plan.
+ */
+export const PAID_BLOCKS_LIST = [
+	'core/cover',
+	'core/video',
```

Comment:
Will this include all paid blocks, like the Payments ones? Or is this only handling core blocks?

Reply:
> Sometimes it's identifying the child block instead of the parent block. For example, on the Contact block, the children are considered paid but the parent isn't...

It has been fixed (partially) [here](https://github.com/Automattic/jetpack/pull/16495/commits/638a9b419bb81676a2f3a9ed201bb73f52804c4d).

 
> WhatsApp showing the upgrade within preview

✅ 

> Contact Form isn't paid, but the field blocks are

This is a particular block in terms he how it's composed. We need to research and find out the way to add a banner properly. I've removed the banner from the child blocks, though.